plugins/node/instrumentation-fs/src/instrumentation.ts

```diff
@@ -0,0 +1,321 @@
+                endHook(fnName, { args: arguments, span });
+              } catch (e) {
+                instrumentation._diag.error('caught endHook error', e);
+              }
+            }
+            span.end();
+            return cb.apply(this, arguments);
+          }
+        );
+
+        try {
+          return api.context.with(
+            api.trace.setSpan(api.context.active(), span),
+            original,
+            this,
+            ...args
+          );
+        } catch (err) {
+          span.recordException(err);
+          span.setStatus({
+            message: err.message,
+            code: api.SpanStatusCode.ERROR,
+          });
+          if (typeof endHook === 'function') {
+            try {
+              endHook(fnName, { args: args, span });
+            } catch (e) {
+              instrumentation._diag.error('caught endHook error', e);
+            }
+          }
+          span.end();
+          throw err;
+        }
+      } else {
+        // TODO: what to do if we are pretty sure it's going to throw
+        return original.apply(this, args);
+      }
+    };
+  }
+
+  protected _patchSyncFunction<T extends (...args: any[]) => ReturnType<T>>(
+    fnName: string,
+    original: T
+  ): T {
+    const instrumentation = this;
+    return <any>function (this: any, ...args: any[]) {
+      if (isTracingSuppressed(api.context.active())) {
+        // Performance optimization. Avoid creating additional contexts and spans
+        // if we already know that the tracing is being suppressed.
+        return original.apply(this, args);
+      }
+      const { createHook, endHook } =
+        instrumentation.getConfig() as FsInstrumentationConfig;
+      if (typeof createHook === 'function') {
+        if (
+          createHook(fnName, {
+            args: args,
+          }) === false
+        ) {
+          // return original.apply(this, args);
+          return api.context.with(
+            suppressTracing(api.context.active()),
+            original,
+            this,
+            ...args
+          );
+        }
+      }
+
+      const span = instrumentation.tracer.startSpan(`fs ${fnName}`) as api.Span;
+      try {
+        // QUESTION: Should we immediately suppress all internal nested calls?
+        const res = api.context.with(
+          api.trace.setSpan(api.context.active(), span),
+          original,
+          this,
+          ...args
+        );
+        if (typeof endHook === 'function') {
+          try {
+            endHook(fnName, { args: args, span });
+          } catch (e) {
+            instrumentation._diag.error('caught endHook error', e);
+          }
+        }
+        span.end();
+        return res;
+      } catch (err) {
+        span.recordException(err);
+        span.setStatus({
+          message: err.message,
+          code: api.SpanStatusCode.ERROR,
+        });
+        if (typeof endHook === 'function') {
+          endHook(fnName, { args: args, span });
```

Comment:
WDYT about also adding the err here to the info? I imagine it can be useful sometimes

Reply:
Added `error`.